### PR TITLE
fix: [N13]: Address licencing issues

### PR DESCRIPTION
**Pull request by chrismaree**
**Motivation**

*OZ identified the following issue:*
```
The following files do not contain an SPDX License identifier:
SpamGuardIdentifierLib.sol
AdminIdentifierLib.sol
Missing license agreements can lead to legal disputes and undesired forms of code usage.
Consider adding a license identifier to all mentioned files.
```

*This PR fixes the issue a requested*

*This PR also fixes this another identified issue through the same set of changes*:
```
The SpamGuardIdentifierLibs.sol and AdminIdentifierLib.sol files do not
define a Solidity version pragma.
It is considered good practice to always specify the desired compiler version by using the
Solidity version pragma. If the contract contains non-trivial operations the resulting bytecode
might differ between Solidity versions including security relevant breaking changes and non-
trivial bug fixes.

Consider defining Solidity version pragmas for all Solidity source files in this project.
```

**Testing**

Check a box to describe how you tested these changes and list the steps for reviewers to test.

- [ ]  Ran end-to-end test, running the code as in production
- [ ]  New unit tests created
- [ ]  Existing tests adequate, no new tests required
- [ ]  All existing tests pass
- [X]  Untested

### CODE REVIEW

File: packages/core/contracts/oracle/implementation/AdminIdentifierLib.sol [hunk 1]
```diff
@@ -1,3 +1,6 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+pragma solidity ^0.8.0;
```
Comment (md0x) [inline packages/core/contracts/oracle/implementation/AdminIdentifierLib.sol:2]:
```suggestion
pragma solidity 0.8.15;
```
Locked solidity version to `0.8.15` [here](https://github.com/UMAprotocol/protocol/pull/4075)

File: packages/core/contracts/oracle/implementation/SpamGuardIdentifierLib.sol [hunk 2]
```diff
@@ -1,3 +1,6 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+pragma solidity ^0.8.0;
```
Comment (md0x) [inline packages/core/contracts/oracle/implementation/SpamGuardIdentifierLib.sol:2]:
```suggestion
pragma solidity 0.8.15;
```
Locked solidity version to `0.8.15` [here](https://github.com/UMAprotocol/protocol/pull/4075)

**Review by md0x — APPROVED** [review]

**Review by mrice32 — APPROVED** [review]
LGTM once @md0x's comments are resolved

File: packages/core/contracts/oracle/implementation/AdminIdentifierLib.sol (hunk 1 above)
Comment (chrismaree) [reply packages/core/contracts/oracle/implementation/AdminIdentifierLib.sol:2]: yup. merging master will fix this. will do now.

**Pull request merged**